mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -445,6 +445,53 @@ impl SwapsContext {
     pub async fn swap_db(&self) -> InitDbResult<SwapDbLocked<'_>> { Ok(self.swap_db.get_or_initialize().await?) }
 }
 
+#[derive(Debug, Deserialize)]
+pub struct GetLockedAmountReq {
+    coin: String,
+}
+
+#[derive(Serialize)]
+pub struct GetLockedAmountResp {
+    coin: String,
+    locked_amount: MmNumberMultiRepr,
+}
+
+#[derive(Debug, Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum GetLockedAmountRpcError {
+    #[display(fmt = "No such coin: {}", coin)]
+    NoSuchCoin { coin: String },
+}
+
+impl HttpStatusCode for GetLockedAmountRpcError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            GetLockedAmountRpcError::NoSuchCoin { .. } => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+}
+
+impl From<CoinFindError> for GetLockedAmountRpcError {
+    fn from(e: CoinFindError) -> Self {
+        match e {
+            CoinFindError::NoSuchCoin { coin } => GetLockedAmountRpcError::NoSuchCoin { coin },
+        }
+    }
+}
+
+pub async fn get_locked_amount_rpc(
+    ctx: MmArc,
+    req: GetLockedAmountReq,
+) -> Result<GetLockedAmountResp, MmError<GetLockedAmountRpcError>> {
+    let _coin = lp_coinfind_or_err(&ctx, &req.coin).await?;
```

Comment:
I think it's worth removing the unused variable declaration and just calling the function.
```rust
lp_coinfind_or_err(&ctx, &req.coin).await?;
```
Or is there some reason why it should be left as it is?